doc/source/providers/amazon.rst

```diff
@@ -149,5 +149,457 @@ On completion, you should receive an ID for your transaction.
 Create Virtual Machine
 ----------------------
 
+This section will guide you through the use of the client to create a
+new Amazon cloud virtual machine in ManageIQ. It assumes you already
+know your Amazon details.
+
+Below are two ways you can use the client to create a virtual machine.
+
+Command Line
+++++++++++++
+
+Command used to create a virtual machine:
+
+.. code-block:: bash
+    :linenos:
+
+    (miq-client) $ miqcli provision_requests create
+
+You can view the available options by passing the **help** option.
+
+.. code-block:: bash
+    :linenos:
+
+    (miq-client) $ miqcli provision_requests create --help
+
+This command when run will send a request to ManageIQ for a new transaction to
+create a new virtual machine. Before running the command, replace the **<<value>>**
+with your Amazon Instance details in the json payload file or payload string.
+
+Example Payloads:
+
+JSON Payload file Auto-placement::
+
+    {
+      "email": "<email_address>",
+      "image": "<image_name>",
+      "flavor": "<instance_type>",
+      "key_pair": "<key_pair>",
+      "vm_name": "<instance_name>",
+      "provider": "<provider_name>",
+      "auto_placement": "true"
+    }
+
+JSON Payload file specifying placement::
+
+    {
+      "email": "<email_address>",
+      "image": "<image_name>",
+      "flavor": "<instance_type>",
+      "key_pair": "<key_pair>",
+      "vm_name": "<instance_name>",
+      "security_group": "<security_group>",
+      "network": "<virtual_private_cloud>",
+      "subnet": "<cloud_subnet>"
+    }
+
+Command with payload given as file
+
+.. code-block:: bash
+    :linenos:
+
+    (miq-client) $ miqcli provision_requests create \
+    --provider Amazon
+    --payload_file <json-payload-file>
+
+Command with payload given as string
+
+.. code-block:: bash
+    :linenos:
+
+    (miq-client) $ miqcli provision_requests create \
+    --provider Amazon
+    --payload "{'email':'<email_address>','image':'<image_name>',
+    'flavor':'<instance_type>','key_pair':'<key_pair>',
+    'vm_name':'<instance_name>','security_group':'<security_group>',
+    'network':'<virtual_private_cloud>','subnet':'<cloud_subnet>'}"
+
+On completion, you should receive an ID for your transaction.
+
+.. code-block:: bash
+
+    INFO: Attempt to create a provision request
+    INFO: Provisioning request created: 566
+
+Use 'miqcli request_tasks status <ID>' to check status of transaction completion.
+Once complete you can use "instances query" to obtain information about the
+instance created.
+
+Python Code
++++++++++++
+
+This code example will demonstrate how you can use the client in a Python
+module to create a virtual machine. Before running the Python code, replace the
+**<value>** with your Amazon Instance details in a json payload file. See
+command line section above for payload file content examples.
+
+.. code-block:: python
```

Comment:
Instead of adding the python code block which looks to be a mirror of the `manageiq-cli/examples/aws_provision_example.py`. We could use [literalinclude](http://www.sphinx-doc.org/en/stable/markup/code.html#directive-literalinclude). to include the source code directly in the docs. This way updates to the source code are reflected to the docs pages. What do you think?